DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
+clientConfig.connectionStrategyConfig.asyncStart = false;
+clientConfig.connectionStrategyConfig.reconnectMode = Config.ReconnectMode.ON;
+```
 
-## 5.8. Enabling Hazelcast Cloud Discovery
+## 6.1. Configuring Client Connection Retry
 
-The purpose of Hazelcast Cloud Discovery is to provide the clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to `true`.
+When client is disconnected from the cluster, it searches for new connections to reconnect.
+You can configure the frequency of the reconnection attempts and client shutdown behavior using
+`ConnectionRetryConfig` (programmatical approach) or `connectionRetry` element (declarative approach).
 
-The following are example configurations.
+Below are the example configurations for each.
 
 **Declarative Configuration:**
 
 ```json
 {
- "group": {
-        "name": "hazel",
-        "password": "cast"
-    },
-
-    "network": {
-        "hazelcastCloud": {
-            "discoveryToken": "EXAMPLE_TOKEN",
-            "enabled": true
+    "connectionStrategy": {
+        "asyncStart": false,
+        "reconnectMode": "ON",
+        "connectionRetry": {
+            "initialBackoffMillis": 1000,
+            "maxBackoffMillis": 60000,
+            "multiplier": 2,
+            "clusterConnectTimeoutMillis": 50000,
+            "jitter": 0.2
         }
     }
 }
-
 ```
 
 **Programmatic Configuration:**
 
 ```javascript
 var clientConfig = new Config.ClientConfig();
-clientConfig.groupConfig.name = 'hazel';
-clientConfig.groupConfig.password = 'cast';
+var connectionRetryConfig = new Config.ConnectionRetryConfig();
+connectionRetryConfig.initialBackoffMillis = 1000;
+connectionRetryConfig.maxBackoffMillis = 60000;
+connectionRetryConfig.multiplier = 2;
+connectionRetryConfig.clusterConnectTimeoutMillis = 50000;
+connectionRetryConfig.jitter = 0.2;
 
-clientConfig.networkConfig.cloudConfig.enabled = true;
-clientConfig.networkConfig.cloudConfig.discoveryToken = 'EXAMPLE_TOKEN';
+clientConfig.connectionStrategyConfig.connectionRetryConfig = connectionRetryConfig;
 ```
 
-To be able to connect to the provided IP addresses, you should use secure TLS/SSL connection between the client and members. Therefore, you should set an SSL configuration as described in the previous section.
+The following are configuration element descriptions:
 
-# 6. Securing Client Connection
+* `initialBackoffMillis`: Specifies how long to wait (backoff), in milliseconds, after the first failure before retrying. Its default value is 1000 ms.
+* `maxBackoffMillis`: Specifies the upper limit for the backoff in milliseconds. Its default value is 30000 ms.
+* `multiplier`: Factor to multiply the backoff after a failed retry. Its default value is 1.
+* `clusterConnectTimeoutMillis`: Timeout value in milliseconds for the client to give up to connect to the current cluster Its default value is 20000.
```

Comment:
nit: change `to the current cluster Its default value is 20000.` to `to the current cluster. Its default value is 20000 ms.`